src/python_bindings/py_util/logging.cpp

```diff
@@ -0,0 +1,228 @@
+
+        if (!payload.empty() && payload.back() == '\n') {
+            payload.remove_suffix(1);
+        }
+
+        (void)py_logger_.attr("log")(level, std::string(payload));
+    }
+
+    void flush_() override {}
+
+private:
+    py::object py_logger_;
+};
+
+using PythonSink = PythonLoggerSink<std::mutex>;
+
+void SetCppLoggingLevel(spdlog::level::level_enum level) {
+    if (auto const logger = ::util::logging::GetLogger()) {
+        logger->set_level(level);
+    }
+}
+
+spdlog::level::level_enum PyLevelToSpdlogLevel(int py_level) {
+    if (py_level > 0 && py_level <= kPyTraceLevel) return spdlog::level::trace;
+    if (py_level > 0 && py_level <= 10) return spdlog::level::debug;
+    if (py_level > 0 && py_level <= 20) return spdlog::level::info;
+    if (py_level > 0 && py_level <= 30) return spdlog::level::warn;
+    if (py_level > 0 && py_level <= 40) return spdlog::level::err;
+    if (py_level > 0) return spdlog::level::critical;
+    return spdlog::level::off;
+}
+
+int SpdlogLevelToPyLevel(spdlog::level::level_enum level) {
+    switch (level) {
+        case spdlog::level::trace:
+            return kPyTraceLevel;
+        case spdlog::level::debug:
+            return 10;
+        case spdlog::level::info:
+            return 20;
+        case spdlog::level::warn:
+            return 30;
+        case spdlog::level::err:
+            return 40;
+        case spdlog::level::critical:
+            return 50;
+        case spdlog::level::off:
+            return 60;
+        default:
+            return 0;
+    }
+}
+
+struct LoggerProxy {
+    py::object real_logger;
+};
+
+void SetupLoggingBridge() {
+    auto python_sink = std::make_shared<PythonSink>();
+    ::util::logging::Initialize("desbordante", {python_sink});
+
+    try {
+        py::gil_scoped_acquire gil;
+        py::module_ logging = py::module_::import("logging");
+        py::object py_logger = logging.attr("getLogger")("desbordante");
+        if (py::len(py_logger.attr("handlers")) == 0) {
+            py::object handler = logging.attr("StreamHandler")();
+            py::object formatter = logging.attr("Formatter")("%(message)s");
+            (void)handler.attr("setFormatter")(formatter);
+            (void)handler.attr("setLevel")(logging.attr("NOTSET"));
+            (void)py_logger.attr("addHandler")(handler);
+            (void)py_logger.attr("setLevel")(logging.attr("INFO"));
+            py_logger.attr("propagate") = false;
+        }
+
+    } catch (py::error_already_set const& e) {
+        py::print("ERROR: Error during Python logging setup:", e.what());
+    }
+}
+
+void CleanupAtExit() {
+    g_is_shutting_down = true;
+    g_proxy_instance = py::none();
+    spdlog::shutdown();
+}
+
+}  // namespace
+
+namespace python_bindings {
+
+void BindLogging(py::module_& main_module) {
+    auto m = main_module.def_submodule("logging",
+                                       "Logging configuration for the desbordante library.");
+
```

Comment:
Why not do this instead:
1) logging.getLogger("desbordante")
2) Construct PythonSink with this object
3) Initialize spdlog
4) Set the minimum logging level in the appropriate places so all messages go to the sink

In PythonSink's `sink_it_`, check the current level of the Python logger (`.level` attribute), and don't format and call its `log` method if it's too high.

You won't need to actually bind anything in that case and can do it in a separate function in bindings.cpp